packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+# Tag 🏷️
+
+These are set of interactive keywords that help organise & categorise objects. Tags can be added or removed from an object by the users.
+
+<img width="426" alt="image" src="https://github.com/razorpay/blade/assets/30949385/75d8068d-d6fc-47e7-aa75-21686ed1c27d">
+
+## Design
+
+- [Tags - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=29210-567576&t=HLRjz3OTofBtFdDl-0)
+- [Tags - Motion](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=29744-568562&mode=design&t=KI2KRJMzUpz8Ymr8-0)
+
+## Proposed API
+
+```jsx
+<Tag
+  size="medium"
+  onDismiss={({ value }) => {
+    console.log('Close Icon Clicked', value);
+  }}
+>
+  Unpaid
+</Tag>
+```
+
+### Props
+
+```ts
+type TagProps = {
+  /**
+   * Decides the size of Tag
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Callback when close icon on Tag is clicked
+   *
+   * Gives a `value` argument which has same text as `children`
+   */
+  onDismiss?: ({ value }: { value: TagProps['children'] }) => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: ReactText;
```

Comment:
Slight difference, we pass the text value as `children` but the `onDismiss` callback receives `value` as a parameter. Should we consider `value` prop instead?

On a related note there are sometimes cases where people might want a different label and value so maybe we do need a `value` prop for such cases (eg. combining with text inputs) 🤔 

```tsx
<Tag value="red">Red color</Tag> // --> shows as Red color in the UI but `red` for form submissions, etc.

<Tag value="red">Red</Tag> // --> just capitalizing 
```

Reply:
I think we can support `value` as optional parameter. When value doesn't exist, the default value will just be the children string in  `value` of `onDismiss` 🤔

Reply:
Added more thoughts here - https://github.com/razorpay/blade/pull/1315#discussion_r1247728513

Regarding form submission, I think that won't be the case since tags are not input elements so they are irrelevant to forms (one of the reasons why I wanted to avoid `value` term)

Reply:
Why are we even going `children` way if JSX is not allowed as children? Generally, the convention is that "children" is used where we have the freedom to render anything, since only text is being allowed I guess passing that via children would be confusing. I agreed with @saurabhdaware that having value since it's not a direct input component won't be a good idea, we can use text prop.

Reply:
I think `value` should suffice for the usecase I'd in mind.

Reply:
We shouldn't have two ways of passing the same data, for such small & simple use cases, if a prop can suffice our use case then no point in supporting children here

Reply:
> Why are we even going children way if JSX is not allowed as children? Generally, the convention is that "children" is used where we have the freedom to render anything, since only text is being allowed I guess passing that via children would be confusing

Fair point. We use children because in cases like Badge, Tags, the text is semantically inside the visual part and it doesn't have a separate description or other components inside of it.

Don't really have much of a strong opinion though. Agree with your reasoning as well that children is usually more flexible. Our children btw is not string but ReactText so I do feel there are some cases where the code might get easier. E.g. -

```jsx
<Badge>{someBadgePrefixTitle} - {anotherBadgeTitle}</Badge>
// vs
<Badge title={`${someBadgePrefixTitle} - ${anotherBadgeTitle}`} />

```

Either ways we will have to keep it children to be consistent with Badge. Can look into changing it on all components from children to some other jsx prop if this is causing confusion?